lib/ansible/module_utils/azure_rm_common.py

```diff
@@ -415,8 +415,8 @@ def get_resource_group(self, resource_group):
         '''
         try:
             return self.rm_client.resource_groups.get(resource_group)
-        except CloudError:
-            self.fail("Parameter error: resource group {0} not found".format(resource_group))
+        except CloudError as cloud_error:
+            self.fail("Error retrieving resource group {0} - {1}".format(resource_group, cloud_error.message))
```

Comment:
Is there a reason why you removed the resource group not found part of the message, are these other errors that `CloudError` would expose?

Reply:
I think the origin error message is too subjective. There could be other reasons causing the exception. 

Yes, I think `CloudError` can give us accurate error message. With the new change, the error is as below if using a non-existing resource group. 


```
TASK [create storage account for function apps] *********************************************************************************************
fatal: [localhost]: FAILED! => {"changed": false, "msg": "Error retrieving resource group zhijzhao4 - Resource group 'zhijzhao4' could not be found."}
```

Reply:
Sounds good to me, just wanted to check.